stl/inc/future

```diff
@@ -927,18 +924,13 @@ private:
     using _Mybase = _State_manager<_Ty*>;
 
 public:
-    future() noexcept {}
+    future() = default;

```

Comment:
This is a terrifying change; however, I _do_ think it's fine, since `future` will never be trivially default constructible..? Same for the destructor.

Reply:
Changing copy ctor triviality affects ABI (as infamously encountered in `tuple<>`) - but that doesn't apply here, so I think you're correct.